### PR TITLE
Add option --override-repository

### PR DESCRIPTION
**Pull request by juj**
Add option --override-repository to allow controlling where git clones happen from.

With this method we can integrate emsdk builds to our CI.

### CODE REVIEW

**Comment by juj** [comment]
Looks like flake8 is run using python2?

**Comment by sbc100** [comment]
Yes, emsdk, unlike emscripten itself still needs to support python2 I believe so that it can bootstrap itself successfully on macOS where python2 comes pre-installed in the path but not python3.

File: emsdk.py [hunk 1]
```diff
@@ -2900,6 +2936,14 @@ def extract_bool_arg(name):
       return True
     return False
 
+  def extract_string_arg(name):
+    nonlocal args
+    for i in range(len(args)):
+      if args[i] == name:
+        value = args[i + 1]
+        args = args[:i] + args[i + 2:]
```
Comment (sbc100) [inline emsdk.py:2944]: Rather then re-assigning `args` you can maybe avoid the `nonlocal` thing by doing `del args[i:i+1]` which is in-place.

File: emsdk.py [hunk 2]
```diff
@@ -182,6 +182,32 @@ def emsdk_path():
...
+      raise Exception('Malformed git URL and refspec ' + url + '!')
+    return url.split('/commit/')
+  else:
+    return (url, 'main')  # Assume the default branch is main in the absence of a refspec
+
+
+emscripten_config_directory = os.path.expanduser("~/")
+# If .emscripten exists, we are configuring as embedded inside the emsdk directory.
+if os.path.exists(os.path.join(emsdk_path(), '.emscripten')):
+  emscripten_config_directory = emsdk_path()
```
Comment (sbc100) [inline emsdk.py:209]: emscripten_config_directory doesn't looked like it is used anywhere in this patch. 

Also there already exists `dot_emscripten_path()` function in this file and we transitioned to only supporting embedded configs.   It doesn't stop you using your own config file, but it just means that emsdk itself will only ever write an embedded config.

Comment (juj) [reply emsdk.py:209]: Ops sorry, `emscripten_config_directory` was a leftover from the previous version of this code, removed it now.

**Comment by juj** [comment]
Updated according to review, ptal.

File: emsdk.py [hunk 3]
```diff
@@ -1684,7 +1709,9 @@ def __init__(self, data):
       setattr(self, key, value)
 
     # Cache the name ID of this Tool (these are read very often)
-    self.name = self.id + '-' + self.version
+    self.name = self.id
+    if len(self.version) > 0:
```
Comment (sbc100) [inline emsdk.py:1713]: Just `if self.version:` is probably fine here?

**Review by sbc100 — APPROVED** [review]
lgtm % test failiures.

**Pull request merged**